fsw/cfe-core/src/inc/cfe_es_table.h

```diff
@@ -0,0 +1,35 @@
+#ifndef cfe_es_table_h
+#define cfe_es_table_h
+
+#include "cfe_platform_cfg.h"
+#include "osconfig.h"
+
+typedef struct
+{
+    char FileName[OS_MAX_PATH_LEN]; /**< \cfetlmmnemonic \ES_APPFILENAME \brief The filename of the file containing the library */
+    char EntryPoint[OS_MAX_API_NAME]; /**< \cfetlmmnemonic \ES_APPENTRYPT \brief The entry point label for the library */
+    char Name[OS_MAX_API_NAME]; /**< \cfetlmmnemonic \ES_APPNAME \brief The registered name of the library */
+}
+CFE_ES_LibTblEntry_t;
+
+typedef struct
+{
+    char FileName[OS_MAX_PATH_LEN]; /**< \cfetlmmnemonic \ES_APPFILENAME \brief The filename of the file containing the application */
+    char EntryPoint[OS_MAX_API_NAME]; /**< \cfetlmmnemonic \ES_APPENTRYPT \brief The entry point label for the application */
+    char Name[OS_MAX_API_NAME]; /**< \cfetlmmnemonic \ES_APPNAME \brief The registered name of the application */
+    uint32 StackSize; /**< \cfetlmmnemonic \ES_STACKSIZE */
+    uint16 Priority; /**< \cfetlmmnemonic \ES_PRIORITY \brief The priority of the application */
+    uint8 ExceptionAction; /**< \cfetlmmnemonic \ES_EXCEPTNACTN \brief What should occur if application has an exception (restart application or restart processor) */
+}
+CFE_ES_AppTblEntry_t;
+
+typedef struct
+{
+    uint32 LibCnt; /**< \brief number of entries in the Libs array */
+    CFE_ES_LibTblEntry_t Libs[CFE_PLATFORM_ES_MAX_LIBRARIES];
+    uint32 AppCnt; /**< \brief number of entries in the Apps array */
+    CFE_ES_AppTblEntry_t Apps[CFE_PLATFORM_ES_MAX_APPLICATIONS];
```

Comment:
Rather than making some one specifically indicate (and maintain) a "count" -- can we just check if the fields are not empty?  I'd suggest looking at the name and/or entry point.  Filename could be validly blank for statically-linked entities that are started by the script.  But every valid entry should have the name and entry point set to a non-empty string.

Reply:
For some reason I just can't get this to work with the static initialization...Can you gin up sample code that works? It just insists on reporting:

```
/home/cdknight/cFS/sample_defs/tables/cpu1_cfe_es_start.c:46:5: warning: missing braces around initializer [-Wmissing-braces]
     {
     ^
/home/cdknight/cFS/sample_defs/tables/cpu1_cfe_es_start.c:46:5: note: (near initialization for ‘CFE_ES_AppLibTbl.Libs’)
/home/cdknight/cFS/sample_defs/tables/cpu1_cfe_es_start.c:55:9: warning: missing braces around initializer [-Wmissing-braces]
         {"SAMPLE_APP", "/cf/sample_app.so", "SAMPLE_AppMain", 16384, 50, 0},
 
```

The code/table in the pull request worked fine.

Reply:
(note also that I've moved the "Name" field to the first field of the struct in my code that generates that gcc warning)

Reply:
You'll still need one more set of braces for the array.  Preference should be to use C99 initializers, to insulate the code from future changes in the table definition.  That is, something like:

```
CFE_ES_AppLibTbl_t CFE_ES_AppLibTbl = 
{
    .Libs = 
    {
        {
            .FileName = "/cf/sample_lib.so", 
            .EntryPoint = "SAMPLE_LibInit", 
            .Name = "SAMPLE_LIB"
        },
    },
    .Apps = 
    {
        /** { <shared library file path>, <entry point>, <name>, <stack size>, <priority>,
                <1 == reset processor on err> } **/
        {
             .FileName = "/cf/sample_app.so", 
             .EntryPoint = "SAMPLE_AppMain", 
             .Name = "SAMPLE_APP", 
             .StackSize = 16384, 
             .Priority = 50, 
             .ExceptionAction =  0
        },
       /* and so on ... */
    }
};
```

This is not only clearer to read, it creates independence in the order of fields between the table definition and the instantiation.